docs/vgclog.md

```diff
@@ -0,0 +1,423 @@
+
+```
+
+Finally, the following example from a balanced policy garbage collection log shows the start of a [concurrent global marking collection](gc.md#garbage-collection-policies), which can be identified by either:
+
+- Searching for the `<concurrent-start>` tag, determining the `contextid` value and searching backwards in the log for the gc cycle with an `id` value that matches this `contextid` value.
+
+- Searching for `type="global mark phase"` to locate the GC cycle for the global marking collection. 
+
+```   
+<exclusive-start id="345" timestamp="2020-11-13T06:32:27.347" intervalms="494.235"> 
+
+<response-info timems="3.588" idlems="1.693" threads="3" lastid="000000000074FF00" lastname="RunDataWriter.1" /> 
+
+</exclusive-start> 
+
+<allocation-taxation id="346" taxation-threshold="402653184" timestamp="2020-11-13T06:32:27.348" intervalms="494.037" /> 
+
+<cycle-start id="347" type="global mark phase" contextid="0" timestamp="2020-11-13T06:32:27.348" intervalms="55328.929" /> 
+
+<gc-start id="348" type="global mark phase" contextid="347" timestamp="2020-11-13T06:32:27.348"> 
+
+<mem-info id="349" free="1147142144" total="3221225472" percent="35"> 
+
+<remembered-set count="1523648" freebytes="122683648" totalbytes="128778240" percent="95" regionsoverflowed="5" regionsstable="321" regionsrebuilding="0"/> 
+
+</mem-info> 
+
+</gc-start> 
+
+<gc-op id="350" type="mark increment" timems="49.623" contextid="347" timestamp="2020-11-13T06:32:27.398"> 
+
+<trace-info objectcount="3117114" scancount="3048551" scanbytes="83420400" /> 
+
+</gc-op> 
+
+<gc-end id="351" type="global mark phase" contextid="347" durationms="49.866" usertimems="344.000" systemtimems="40.000" stalltimems="9.352" timestamp="2020-11-13T06:32:27.398" activeThreads="8"> 
+
+<mem-info id="352" free="1147142144" total="3221225472" percent="35"> 
+
+<remembered-set count="1768768" freebytes="121703168" totalbytes="128778240" percent="94" regionsoverflowed="1" regionsstable="0" regionsrebuilding="326"/> 
+
+</mem-info> 
+
+</gc-end> 
+
+<concurrent-start id="353" type="GMP work packet processing" contextid="347" timestamp="2020-11-13T06:32:27.399"> 
+
+<concurrent-mark-start scanTarget="113512867" /> 
+
+</concurrent-start> 
+
+<exclusive-end id="354" timestamp="2020-11-13T06:32:27.399" durationms="52.216" /> 
+
+ 
+ 
+
+<concurrent-end id="355" type="GMP work packet processing" contextid="347" timestamp="2020-11-13T06:32:27.538"> 
+
+<concurrent-mark-end bytesScanned="113629324" reasonForTermination="Work target met" /> 
+
+</concurrent-end>
+```
+
+You can determine the following features by analyzing this portion of the log: 
+
+- The `<exclusive-start>` tag precedes the`<cycle-start>` tag, indicating that the concurrent global marking starts with a *stop-the-world* event.
+
+- The *stop-the-world* event consists of two operations; the mark increment operation and mark phase operation. After these 2 GC operations, a GMP work processing event starts as a concurrent event.
```

Comment:
Confusing - I was looking for 2 `<gc-op>` events for mark increment and mark phase, but the mark phase seems to be an attribute of the `<gc-start>` and `<gc-end>` elements instead. I wonder whether we could add the line to look for in brackets as part of the explanation so that users can refer back and forth?

Can we also explain the outcome of this global mark phase example in terms of memory? Looks to be something to do with the values on `<remembered-set>` that reflects "Work target met"?